builtin/providers/aws/resource_aws_elasticache.go

```diff
@@ -0,0 +1,256 @@
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set: func(v interface{}) int {
+					return hashcode.String(v.(string))
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsElasticacheCreate(d *schema.ResourceData, meta interface{}) error {
+	elasticacheconn := meta.(*AWSClient).elasticacheconn
+
+	clusterId := d.Get("cluster_id").(string)
+	nodeType := d.Get("node_type").(string)           // e.g) cache.m1.small
+	numNodes := d.Get("num_cache_nodes").(int)        // 2
+	engine := d.Get("engine").(string)                // memcached
+	engineVersion := d.Get("engine_version").(string) // 1.4.14
+	port := d.Get("port").(int)                       // 11211
+	subnetGroupName := d.Get("subnet_group_name").(string)
+	securityNameSet := d.Get("security_group_names").(*schema.Set)
+	securityIdSet := d.Get("security_group_ids").(*schema.Set)
+	paramGroupName := d.Get("parameter_group_name").(string) // default.memcached1.4
+
+	securityNames := make([]string, securityNameSet.Len())
+	for i, name := range securityNameSet.List() {
+		securityNames[i] = name.(string)
+	}
+	securityIds := make([]string, securityIdSet.Len())
+	for i, id := range securityIdSet.List() {
+		securityIds[i] = id.(string)
+	}
+
+	req := &elasticcache.CreateCacheClusterMessage{
+		CacheClusterID:          aws.String(clusterId),
+		CacheNodeType:           aws.String(nodeType),
+		NumCacheNodes:           aws.Integer(numNodes),
+		Engine:                  aws.String(engine),
+		EngineVersion:           aws.String(engineVersion),
+		Port:                    aws.Integer(port),
+		CacheSubnetGroupName:    aws.String(subnetGroupName),
+		CacheSecurityGroupNames: securityNames,
+		SecurityGroupIDs:        securityIds,
+		CacheParameterGroupName: aws.String(paramGroupName),
+	}
+
+	_, err := elasticacheconn.CreateCacheCluster(req)
+	if err != nil {
+		return fmt.Errorf("Error creating Elasticache: %s", err)
+	}
+
+	d.SetId(clusterId)
+
+	return nil
+}
+
+func resourceAwsElasticacheRead(d *schema.ResourceData, meta interface{}) error {
+	elasticacheconn := meta.(*AWSClient).elasticacheconn
+	req := &elasticcache.DescribeCacheClustersMessage{
+		CacheClusterID: aws.String(d.Id()),
+	}
+
+	res, err := elasticacheconn.DescribeCacheClusters(req)
+	if err != nil {
+		return err
+	}
+
+	if len(res.CacheClusters) == 1 {
+		c := res.CacheClusters[0]
+		d.Set("cluster_id", c.CacheClusterID)
+		d.Set("node_type", c.CacheNodeType)
+		d.Set("num_cache_nodes", c.NumCacheNodes)
+		d.Set("engine", c.Engine)
+		d.Set("engine_version", c.EngineVersion)
+		if c.ConfigurationEndpoint != nil {
+			d.Set("port", c.ConfigurationEndpoint.Port)
+		}
+		d.Set("subnet_group_name", c.CacheSubnetGroupName)
+		d.Set("security_group_names", c.CacheSecurityGroups)
+		d.Set("security_group_ids", c.SecurityGroups)
+		d.Set("parameter_group_name", c.CacheParameterGroup)
+	}
+
+	return nil
+}
+
+func resourceAwsElasticacheUpdate(d *schema.ResourceData, meta interface{}) error {
+	return nil
+}
```

Comment:
Heh it's true we should eventually do that - I just didn't want to force a bunch of extra work on our friend @tmtk75 since he's put in so much already.

We've generally seen a pattern of (a) initial impl with `ForceNew` everywhere and no update, (b) follow on with update support.

But that's not saying I'd complain if the implementation showed up. :wink: